File: app/src/main/java/com/example/david/lists/widget/configview/IWidgetConfigContract.java
package com.example.david.lists.widget.configview;

import com.example.david.lists.data.datamodel.UserList;

import java.util.List;

public interface IWidgetConfigContract {
    interface View {
        void setData(List<UserList> list);

        void setStateDisplayList();

        void setStateLoading();

        void setStateError(String message);

        void setResults(int widgetId, int resultCode);

        void finishView(int widgetId);

        void finishViewInvalidId();
    }

    interface Adapter {
        void setData(List<UserList> list);
    }

    interface Logic {
        void onStart(int appWidgetId);

        void selectedUserList(UserList userList);

        void onDestroy();
    }

    interface ViewModel {
        void setViewData(List<UserList> list);

        List<UserList> getViewData();

        void setWidgetId(int widgetId);

        int getWidgetId();

        int getInvalidWidgetId();

        int getResultOk();

        int getResultCancelled();

        String getErrorMsg();

        String getErrorMsgEmptyList();

        String getSharedPrefKeyId();

        String getSharedPrefKeyTitle();
    }
}

File: app/src/main/java/com/example/david/lists/widget/configview/IWidgetConfigContract.kt
package com.example.david.lists.widget.configview

import com.example.domain.datamodel.UserList

interface IWidgetConfigContract {
    interface View {
        fun setViewData(list: List<UserList>)

        fun setStateDisplayList()

        fun setStateLoading()

        fun setStateError(message: String)

        fun setResults(widgetId: Int, resultCode: Int)

        fun saveDetails(id: String, title: String, sharedPrefKeyId: String, sharedPrefKeyTitle: String)

        fun finishView(widgetId: Int)

        fun finishViewInvalidId()
    }

    interface Adapter {
        fun setData(list: List<UserList>)
    }

    interface Logic {
        fun onStart(widgetId: Int)

        fun selectedUserList(position: Int)

        fun onDestroy()
    }

    interface ViewModel {
        var viewData: List<UserList>

        var widgetId: Int

        val invalidWidgetId: Int

        val resultOk: Int

        val resultCancelled: Int

        val errorMsg: String

        val errorMsgEmptyList: String

        val sharedPrefKeyId: String

        val sharedPrefKeyTitle: String
    }
}
